### DISCUSSION TITLE
Multiple Package type support(gradle/Pakage.json)

### DISCUSSION THREAD

**Discussion by skagithala-clgx** [paketo-buildpacks/feedback#260] [Java Team]
Hi Team,

We have use case where multistack repo is configured and it is using gradle with packeto build pack and it is using common folder for dependencies resolutions. trying to install npm_install, failing with below errors
Issue:
it is unable to install npm_install

> Task :frontend:npm_install
18:11:30        Caching disabled for task ':frontend:npm_install' because:
18:11:30          Build cache is disabled
18:11:30        Task ':frontend:npm_install' is not up-to-date because:
18:11:30          Task has not declared any outputs despite executing actions.
18:11:30        Starting process 'command '/workspace/frontend/.gradle/npm/npm-v6.14.10/bin/npm''. Working directory: /workspace/frontend Command: /workspace/frontend/.gradle/npm/npm-v6.14.10/bin/npm install
18:11:30        Successfully started process 'command '/workspace/frontend/.gradle/npm/npm-v6.14.10/bin/npm''
18:11:38        npm WARN tar ENOENT: no such file or directory, open '/workspace/frontend/node_modules/.staging/@schematics/update-38800c5d/LICENSE'
18:11:38        npm WARN tar ENOENT: no such file or directory, open '/workspace/frontend/node_modules/.staging/core-js-ea45384e/proposals/collection-methods.js'
18:11:38        npm WARN tar ENOENT: no such file or directory, open '/workspace/frontend/node_modules/.staging/yaml-f611cfa4/dist/index.js'

below are the configurations
settings.gradle
rootProject.name = 'xyz'
include 'frontend'
include 'backend'
In build.gradle of project1 below line is mentioned for dependency

frontend/build.gradle contains  buildFrontend.dependsOn npm_install
Backend/build.gradle contains
bootJar.dependsOn ':frontend:npm_install'
bootJar.dependsOn ':frontend:buildFrontend'

Project structure: 
project1
project2
gradlew
gradlew.bat
settings.gradle

Could you please guide on mono repo configuration, provide if we have any documentation for the same.

**Comment by dmikusa** [comment]
Sorry, I'm not sure what would cause those files to be missing, but I would suggest you check out our samples. We have a sample Gradle + Node.js project here -> https://github.com/paketo-buildpacks/samples/tree/main/java/java-node/gradle-node Perhaps that will give you some clues on how to get your custom project working.

**Reply by skagithala-clgx** [reply]
Thanks for the inputs 
do we have sample repos for multi module gradle & maven, monorepo for gradel
